daml-lf/spec/daml-lf-1.rst

```diff
@@ -3253,65 +3191,74 @@ as described by the ledger model::
        ⇓ᵤ
      (Ok v₁, tr₁) ‖ S₁
 
-     e₁  ⇓  Err @τ₁ v₁
-     e₂[x ↦ 'to_any_exception' @τ₁ v₁]  ⇓  Ok ('None' @σ)
-   —————————————————————————————————————————————————————————————————————— EvUpdTryCatchErr1_NoHandle
-     'try' @τ e₁ 'catch' x. e₂ ‖ S₀  ⇓ᵤ  (Err @τ₁ v₁, ε)
+     e₁  ⇓  Err (Fatal t)
+   —————————————————————————————————————————————————————————————————————— EvUpdTryCatchFatal1
+     'try' @τ e₁ 'catch' x. e₂ ‖ S₀  ⇓ᵤ  (Err (Fatal t), ε)
+
+     e₁  ⇓  Ok u₁
+     u₁ ‖ S₀  ⇓ᵤ  (Err (Fatal t), tr₁)
+   —————————————————————————————————————————————————————————————————————— EvUpdTryCatchFatal2
+     'try' @τ e₁ 'catch' x. e₂ ‖ S₀  ⇓ᵤ  (Err (Fatal t), tr₁)
+
+     e₁  ⇓  Err (Throw v)
+     e₂[x ↦ v]  ⇓  Ok ('None' @σ)
+   —————————————————————————————————————————————————————————————————————— EvUpdTryCatchThrow1_NoHandle
+     'try' @τ e₁ 'catch' x. e₂ ‖ S₀  ⇓ᵤ  (Err (Throw v), ε)
 
      e₁  ⇓  Ok u₁
-     u₁ ‖ S₀  ⇓ᵤ  (Err @τ₁ v₁, tr₁)
-     e₂[x ↦ 'to_any_exception' @τ₁ v₁]  ⇓  Ok ('None' @σ)
-   —————————————————————————————————————————————————————————————————————— EvUpdTryCatchErr2_NoHandle
-     'try' @τ e₁ 'catch' x. e₂ ‖ S₀  ⇓ᵤ  (Err @τ₁ v₁, tr₁)
+     u₁ ‖ S₀  ⇓ᵤ  (Err (Throw v), tr₁)
+     e₂[x ↦ v]  ⇓  Ok ('None' @σ)
+   —————————————————————————————————————————————————————————————————————— EvUpdTryCatchThrow2_NoHandle
+     'try' @τ e₁ 'catch' x. e₂ ‖ S₀  ⇓ᵤ  (Err (Throw v), tr₁)
 
-     e₁  ⇓  Err @τ₁ v₁
-     e₂[x ↦ 'to_any_exception' @τ₁ v₁]  ⇓  Ok ('Some' @σ u₂)
+     e₁  ⇓  Err (Throw v₁)
+     e₂[x ↦ v₁]  ⇓  Ok ('Some' @σ u₂)
      u2 ‖ S₀  ⇓ᵤ  (Ok v₂, tr₂) ‖ S₂
-   —————————————————————————————————————————————————————————————————————— EvUpdTryCatchErr1_OkHandle_Ok
+   —————————————————————————————————————————————————————————————————————— EvUpdTryCatchThrow1_OkHandle_Ok
      'try' @τ e₁ 'catch' x. e₂ ‖ S₀
        ⇓ᵤ
      (Ok v₂, ('rollback' ε) ⋅ tr₂) ‖ S₂
 
      e₁  ⇓  Ok u₁
-     u₁ ‖ S₀  ⇓ᵤ  (Err @τ₁ v₁, tr₁)
-     e₂[x ↦ 'to_any_exception' @τ₁ v₁]  ⇓  Ok ('Some' @σ u₂)
+     u₁ ‖ S₀  ⇓ᵤ  (Err (Throw v₁), tr₁)
+     e₂[x ↦ v₁]  ⇓  Ok ('Some' @σ u₂)
      u2 ‖ S₀  ⇓ᵤ  (Ok v₂, tr₂) ‖ S₂
-   —————————————————————————————————————————————————————————————————————— EvUpdTryCatchErr2_OkHandle_Ok
+   —————————————————————————————————————————————————————————————————————— EvUpdTryCatchThrow2_OkHandle_Ok
      'try' @τ e₁ 'catch' x. e₂ ‖ S₀
        ⇓ᵤ
      (Ok v₂, ('rollback' tr₁) ⋅ tr₂) ‖ S₂
```

Comment:
I guess technically we need to insert normalization here or alternatively describe it as a postprocessing step somewhere.

Reply:
Similarly for the other cases below.

Reply:
I think we should keep it as a separate step, since that's closer to the implementation.